packages/firestore/src/local/indexeddb_query_cache.ts

```diff
@@ -86,6 +86,14 @@ export class IndexedDbQueryCache implements QueryCache {
     });
   }
 
+  getHighestSequenceNumber(
+    transaction: PersistenceTransaction
+  ): PersistencePromise<ListenSequenceNumber> {
+    return getHighestListenSequenceNumber(
+      (transaction as IndexedDbTransaction).simpleDbTransaction
```

Comment:
I think that you can get rid of this cast if you change retrieveMetadata to use an `PersistenceTransaction` and `IndexedDbPersistence.getStore`.

Reply:
`IndexedDbPersistence` also calls `getHighestListenSequenceNumber` from a context in which a `PersistenceTransaction` does not make sense, but a `SimpleDbTRansaction` does. Namely, the `start()` method, where there is no sequence number and no visible-external-to-persistence-layer transaction.